Controllers/BatchController.cs

```diff
@@ -45,81 +45,51 @@ public async Task<IActionResult> Post(BatchRequest request)
             {
                 if (mutation.Id == 0)
                 {
-                    return BadRequest(String.Format("id field of mutation must be non-zero"));
+                    return BadRequest("id field of mutation must be non-zero");
                 }
 
-                var currentMutationID = await db.GetMutationID(accountID, request.ClientId);
-                var expectedMutationID = currentMutationID + 1;
-
-                if (mutation.Id > expectedMutationID)
-                {
-                    return BadRequest(String.Format("Mutation ID {0} is too high - next expected mutation is {1}", mutation.Id, expectedMutationID));
-                }
-
-                if (mutation.Id < expectedMutationID)
-                {
-                    infos.Add(new MutationInfo
-                    {
-                        Id = mutation.Id,
-                        Message = String.Format("Mutation ID {0} has already been processed. Skipping.", mutation.Id),
-                    });
-                    continue;
-                }
-
-                try
+                var result = await ProcessMutation(db, accountID, request.ClientId, mutation);
+                if (result is BadRequestObjectResult)
                 {
-                    await ProcessMutation(db.Items, accountID, mutation);
+                    return result as BadRequestObjectResult;
                 }
-                catch (PermanentError e)
```

Comment:
The definition of PermanentError can be removed (and maybe the comment should be moved into JS?)

Reply:
Good point